lib/pbench/server/filetree.py

```diff
@@ -243,6 +242,59 @@ def create(cls, tarball: Path, controller: "Controller") -> "Tarball":
 
         return cls(destination, controller)
 
+    def extract(self, path: Path) -> str:
+        """
+        Extract a file from the tarball and return it as a string
+
+        TODO: Should we allow fetching a full directory path, e.g., "/" to
+        extract everything? And, if so, in what form would we return it.
+        Similarly, what about a binary file?
```

Comment:
Seems like this would make a good candidate for future work, and track it in an issue.

Reply:
I think I would prefer to see an actual need for this....